src/FloorSubsystem.java

```diff
@@ -0,0 +1,90 @@
+import java.io.BufferedReader;
+import java.io.FileNotFoundException;
+import java.io.FileReader;
+import java.io.IOException;
+import java.text.ParseException;
+import java.time.LocalTime;
+import java.util.ArrayList;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Queue;
+
+/**
+ * The FloorSubsystem Class parses events and processes the information from the
+ * events, sending floorEvents and taking arrival events from scheduler
+ * 
+ */
+public class FloorSubsystem implements Runnable {
+
+	private Queue<FloorEvent> eventList;
+	private List<Floor> floors;
+	private int numOfFloors;
+	private MiddleMan box;
+	private String filename;
+
+	/**
+	 * Constructor for FloorSubsystem
+	 * 
+	 * @param filename    the file to be parsed
+	 * @param numOfFloors the number of floors
+	 * @param box         where events will be put and received
+	 */
+	public FloorSubsystem(String filename, int numOfFloors, MiddleMan box) {
+		this.filename = filename;
+		this.box = box;
+		this.numOfFloors = numOfFloors;
+		this.eventList = new LinkedList<>();
+		this.floors = new ArrayList<>();
```

Comment:
I'm pretty sure you should also initialize all of the floor objects in this constructor and add them into this.floors